### PR TITLE
Feature/kitosudv-2483-task-2643

### PR DESCRIPTION
**Pull request by a-naskret**
## KITOS Pull request template
The following procedure dictates the steps needed before a Pull request can be merged into master.

- [ ] **Implement**: 
      *All requirements are implemented and unit tests are green*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*
      
- [ ] **Request review**: 
      *Tag whomever you wish to review your code*
      
- [ ] **Review completed**: 
      *Reviewer ticks this box when review comments have been submitted*
      
- [ ] **Changes**: 
      *PR owner and reviewer agrees on which changes must be made and the changes are committed.*
      
- [ ] **Merge master into branch / rebase with master**: 
      *Make sure you are testing your changes and how they co-exist with the latest version of master*
      
- [ ] **Green on integration**: 
      *All integration tests are green on integration*

### CODE REVIEW

File: Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+using Presentation.Web.Infrastructure.Attributes;
+using Presentation.Web.Models.API.V1;
+using Presentation.Web.Models.API.V1.UI_Configuration;
+using Swashbuckle.Swagger.Annotations;
+
+namespace Presentation.Web.Controllers.API.V1.UI_Configuration
+{
+    [InternalApi]
+    [RoutePrefix("api/v1/organizations/{organizationId}/ui-config/modules/{module}")]
+    public class UIConfigurationController : BaseApiController
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs:17]: rename to UIModuleconfiguratioController since it is scoped to config of modules :-)

File: Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    {
+        [HttpGet]
+        [Route]
+        [SwaggerResponse(HttpStatusCode.OK, Type = typeof(ApiReturnDTO<CustomizedUINodeDTO>))]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage Get(int organizationId, string module)
+        {
+            return Ok();
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs:26]: that is not a valid empty DTO for a 200 response :-)

File: Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+namespace Presentation.Web.Controllers.API.V1.UI_Configuration
+{
+    [InternalApi]
+    [RoutePrefix("api/v1/organizations/{organizationId}/ui-config/modules/{module}")]
+    public class UIConfigurationController : BaseApiController
+    {
+        [HttpGet]
+        [Route]
+        [SwaggerResponse(HttpStatusCode.OK, Type = typeof(ApiReturnDTO<CustomizedUINodeDTO>))]
+        [SwaggerResponse(HttpStatusCode.Forbidden)]
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs:22]: missing UnAuthorized (if called without cookie)

File: Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        [SwaggerResponse(HttpStatusCode.NotFound)]
+        public HttpResponseMessage Get(int organizationId, string module)
+        {
+            return Ok();
+        }
+
+        [HttpPut]
+        [Route]
+        [SwaggerResponse(HttpStatusCode.NoContent)]
+        [SwaggerResponse(HttpStatusCode.BadRequest)]
```
Comment (mrjsawdk) [inline Presentation.Web/Controllers/API/V1/UI_Configuration/UIConfigurationController.cs:32]: missing UnAuthorized (if called without cookie)

File: Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+using System.Configuration;
+using System.Linq;
+using System.Web;
+
+namespace Presentation.Web.Models.API.V1.UI_Configuration
+{
+    public class CustomizedUINodeDTO
+    {
+        [Required]
+        [RegexStringValidator("^([a-zA-Z]+)(\\.[a-zA-Z]+)*$")]
```
Comment (mrjsawdk) [inline Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs:13]: regex should be extracted to a const so it can be reused in the domain :-)

File: Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+        public string FullKey { get; set; }
```
Comment (mrjsawdk) [inline Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs:14]: perhaps just call it key here.. in the UI FullKey makes sense but here it is just a key... my bad. I will change it in ui

File: Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs [hunk 7]
```diff
@@ -0,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Configuration;
+using System.Linq;
+using System.Web;
+
+namespace Presentation.Web.Models.API.V1.UI_Configuration
+{
```
Comment (mrjsawdk) [inline Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs:9]: we are missing the top level dto .. the one with "nodes". that leaves us the flexibility to add more properties to the root level without breaking the contract with existing bindings

Comment (mrjsawdk) [reply Presentation.Web/Models/API/V1/UI_Configuration/CustomizedUINodeDTO.cs:9]: Remember to align the tests and swagger doc

File: Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+            {
+                new()
+                {
+                    FullKey = key,
+                    Enabled = isEnabled
+                }
+            };
+
+            //Act
+            var response = await UIConfigurationHelper.SendPutRequestAsync(TestEnvironment.DefaultOrganizationId, module, body);
```
Comment (mrjsawdk) [inline Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs:35]: remember `using var`. We don't want handle leaks

File: Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        }
+
+        [Fact]
+        public async Task Can_Get()
+        {
+            //Arrange
+            var module = A<string>();
+
+            //Act
+            var response = await UIConfigurationHelper.SendGetRequestAsync(TestEnvironment.DefaultOrganizationId, module);
```
Comment (mrjsawdk) [inline Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs:48]: missing `using var`
Also you are missing the test where you validate that "nodes" contains an empty, not-null array

**Review by mrjsawdk — COMMENTED** [review]
@a-naskret Minor adjustments
Also - set the merge target as `feature/KITOSUDV-2483` which is the main feature branch for this story

File: Presentation.Web/Models/API/V1/UI_Configuration/UIModuleCustomizationDTO.cs [hunk 10]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace Presentation.Web.Models.API.V1.UI_Configuration
+{
+    public class UIModuleCustomizationDTO
+    {
+        public List<CustomizedUINodeDTO> Nodes { set; get; }
```
Comment (mrjsawdk) [inline Presentation.Web/Models/API/V1/UI_Configuration/UIModuleCustomizationDTO.cs:10]: this can be IEnumerable (don't change in this PR)

File: Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+            //Act
+            using var response = await UIConfigurationHelper.SendGetRequestAsync(TestEnvironment.DefaultOrganizationId, module);
+
+            //Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var data = await response.ReadResponseBodyAsKitosApiResponseAsync<UIModuleCustomizationDTO>();
+            Assert.Empty(data.Nodes);
+            Assert.NotNull(data.Nodes);
```
Comment (mrjsawdk) [inline Tests.Integration.Presentation.Web/UI_Configuration/UIConfigurationTests.cs:58]: hehe the order is off. nulle check before isEmpty.. again just change in the next PR

**Review by mrjsawdk — APPROVED** [review]

**Comment by mrjsawdk** [comment]
@a-naskret I added a few comments to be updated in your next branch
